blockdag/utxodiffstore.go

```diff
@@ -177,12 +185,23 @@ func (diffStore *utxoDiffStore) clearDirtyEntries() {
 
 // dbStoreDiffData stores the UTXO diff data to the database.
 // This overwrites the current entry if there exists one.
-func dbStoreDiffData(dbTx database.Tx, hash *daghash.Hash, diffData *blockUTXODiffData) error {
-	serializedDiffData, err := serializeBlockUTXODiffData(diffData)
+func dbStoreDiffData(dbTx database.Tx, buffer *bytes.Buffer, hash *daghash.Hash, diffData *blockUTXODiffData) error {
+	// To avoid a ton of allocs, reuse the given buffer.
+	// We expect the buffer, in most cases, to already
+	// be large enough to accommodate the serialized data
+	// without growing.
+	bufferFrom := buffer.Len()
```

Comment:
Didn't we say we want to reset the buffer so that we don't need to do the bufferFrom stuff?